src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -285,12 +285,16 @@ private void OnException(HttpContext httpContext, Exception exception)
                 var telemetry = httpContext?.Features.Get<RequestTelemetry>();
                 if (telemetry != null)
                 {
-                    telemetry.Success = false;
+                    if (int.TryParse(telemetry.ResponseCode, out int _responseCode))
```

Comment:
nitpick: per coding standards for C# please don't use underscores as prefixes to variable names.